controllers/provider-azure/pkg/webhook/controlplane/ensurer.go

```diff
@@ -229,3 +238,65 @@ func (e *ensurer) EnsureKubeletCloudProviderConfig(ctx context.Context, ectx gen
 	*data = cm.Data[azure.CloudProviderConfigMapKey]
 	return nil
 }
+
+// EnsureAdditionalFile ensures additional systemd files
+func (e *ensurer) EnsureAdditionalFiles(ctx context.Context, ectx genericmutator.EnsurerContext, files *[]extensionsv1alpha1.File) error {
+	if err := e.ensureAcrConfigFile(ctx, ectx, files); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *ensurer) ensureAcrConfigFile(ctx context.Context, ectx genericmutator.EnsurerContext, files *[]extensionsv1alpha1.File) error {
+	for _, f := range *files {
+		if f.Path == acrConfigPath {
+			return nil
+		}
+	}
+
+	cm, exists, err := e.getAcrConfigMap(ctx, ectx)
+	if err != nil {
+		return nil
+	}
+	if !exists {
+		return nil
+	}
+
+	// Write the content of the file.
+	fciCodec := controlplane.NewFileContentInlineCodec()
+	fci, err := fciCodec.Encode([]byte(cm.Data[azure.CloudProviderAcrConfigMapKey]), string(cloudinit.B64FileCodecID))
+	if err != nil {
+		return errors.Wrap(err, "could not encode acr cloud provider config")
+	}
+
+	*files = append(*files, extensionsv1alpha1.File{
+		Path:        acrConfigPath,
+		Permissions: util.Int32Ptr(0644),
+		Content: extensionsv1alpha1.FileContent{
+			Inline: fci,
+		},
+	})
+	return nil
+}
+
+func (e *ensurer) getAcrConfigMap(ctx context.Context, ectx genericmutator.EnsurerContext) (*corev1.ConfigMap, bool, error) {
```

Comment:
Do you need this `bool`? As far as I can see it is `false` <=> `*corev1.ConfigMap == nil`